packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -71,21 +120,46 @@ class ImageCache {
   ImageStreamCompleter putIfAbsent(Object key, ImageStreamCompleter loader()) {
     assert(key != null);
     assert(loader != null);
-    ImageStreamCompleter result = _cache[key];
-    if (result != null) {
+    final ImageStreamCompleter result = _pendingImages[key];
+    // Nothing needs to be done because the image hasn't loaded yet.
+    if (result != null)
+      return result;
+    final _CachedImage image = _cache.remove(key);
+    if (image != null) {
       // Remove the provider from the list so that we can put it back in below
       // and thus move it to the end of the list.
```

Comment:
the way you've rearranged the code this comment makes no sense now.

Reply:
Fixed